source/special.tex

```diff
@@ -3203,17 +3203,23 @@
 \end{example}
 
 \pnum
-When the criteria for elision of a copy/move operation are met,
-but not for an \nonterminal{exception-declaration},
-and the object
-to be copied is designated by an lvalue,
-or when the \grammarterm{expression} in a \tcode{return} statement
+In the following copy-initialization contexts, a move constructor might be used instead of a copy constructor:
+\begin{itemize}
+\item If the \grammarterm{expression} in a \tcode{return} statement~(\ref{stmt.return})
 is a (possibly parenthesized) \grammarterm{id-expression}
 that names an object with automatic storage duration declared in the body
 or \grammarterm{parameter-declaration-clause} of the innermost enclosing
-function or \grammarterm{lambda-expression},
+function or \grammarterm{lambda-expression}, or
+
+\item if the operand of a \grammarterm{throw-expression}~(\ref{expr.throw})
+is the name of a non-volatile automatic object
+(other than a function or catch-clause parameter)
+whose scope does not extend beyond the end of the innermost enclosing
+\grammarterm{try-block} (if there is one),
```

Comment:
This also seems like a defect (perhaps part of the same one): the *try-block* includes the *handler*s, which means you don't get to do copy-to-move conversion in this case, where it would obviously be safe:

```
void f() {
  try { /*...*/ }
  catch (my_exception a) {
    if (cond) throw a; // no move here =(
    my_wrapper_exception b(a);
    throw b; // no move here either =(
  }
}
```

There doesn't seem to be a technical reason to not move in the above cases.